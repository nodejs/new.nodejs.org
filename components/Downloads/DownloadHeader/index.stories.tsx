import DownloadHeader from '.';
import type { Meta as MetaObj, StoryObj } from '@storybook/react';

type Story = StoryObj<typeof DownloadHeader>;
type Meta = MetaObj<typeof DownloadHeader>;

export const Default: Story = {};

export default { component: DownloadHeader } as Meta;
