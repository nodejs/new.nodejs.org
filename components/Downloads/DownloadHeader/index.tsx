import { FormattedMessage, injectIntl } from 'react-intl';
import styles from './index.module.scss';
import SectionTitle from '../../Common/SectionTitle';
import type { WrappedComponentProps } from 'react-intl';
import type { NodeReleaseData } from '../../../types';

type DownloadHeaderProps = { release: NodeReleaseData }
  release?: NodeReleaseData;
};

const DownloadHeader = ({ release, intl }: Props & WrappedComponentProps) => (
  <>
    <div className={styles.downloadHeader}>
      <SectionTitle
        path={[
          'home',
          intl.formatMessage({
            id: 'components.downloadHeader.navigation.activeSection',
          }),
        ]}
      />
      <div>
        <FormattedMessage
          id="components.downloadHeader.navigation.nodeVersion"
          values={{ lts: release?.isLts, nodeVersion: release?.fullVersion }}
        />
      </div>
    </div>
    <div className={styles.downloadHeader}>
      <div className={styles.title}>
        <FormattedMessage id="components.downloadHeader.navigation.title" />
      </div>
      <div className={styles.npm}>
        <FormattedMessage id="components.downloadHeader.navigation.npmVersion" />
      </div>
    </div>
  </>
);

export default injectIntl(DownloadHeader);
