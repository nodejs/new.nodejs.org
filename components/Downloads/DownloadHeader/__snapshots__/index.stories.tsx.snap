// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Downloads/DownloadHeader Default smoke-test 1`] = `
<div class="DownloadHeader_downloadHeader__Dp85U">
  <div class="SectionTitle_sectionTitle__mNok7">
    home /
    <span class="SectionTitle_active__XRLOn">
      downloads
    </span>
  </div>
  <div>
    CURRENT VERSION v18.0.0
  </div>
</div>
<div class="DownloadHeader_downloadHeader__Dp85U">
  <div class="DownloadHeader_title__KKyS9">
    Downloads
  </div>
  <div class="DownloadHeader_npm__3bjKz">
    (includes npm)
  </div>
</div>
`;
